metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -253,8 +260,16 @@ def __init__(self, ctx: PipelineContext, config: BigQueryV2Config):
 
         # Global store of table identifiers for lineage filtering
         self.table_refs: Set[str] = set()
-        # Maps project -> view_ref -> [upstream_table_ref], for view lineage
-        self.view_upstream_tables: Dict[str, Dict[str, List[str]]] = defaultdict(dict)
+
+        # We do this so that the SQL is stored in a file-backed dict, but the sql IDs are stored in memory.
+        # Maps project -> view_ref -> sql ID (will be used when generating lineage)
+        self.view_definition_ids: Dict[str, Dict[str, str]] = defaultdict(dict)
+        # Maps sql ID -> actual sql
+        self.view_definitions: FileBackedDict[str] = FileBackedDict()
+
+        self.sql_parser_schema_resolver = SchemaResolver(
+            platform=self.platform, env=self.config.env
+        )
```

Comment:
Ideally, I think we'd implement `SchemaResolver.close` and call it when this source is closed and when the datahub graph is closed, rather than relying on garbage collection, which I think is the case right now